src/programs/city/construct.js

```diff
@@ -48,9 +48,9 @@ class CityConstruct extends kernel.process {
       }
       let result = this.room.constructNextMissingStructure()
       this.data.lastscan = Game.time
-      if (Number.isInteger(result) && result < 0) {
-        let type = this.room.getNextMissingStructureType()
-        Logger.log(`Unable to build structure (${type}) in ${this.data.room}: ${result}`, LOG_ERROR)
+      if (result !== false && result[0] < 0) {
+        // result is [code, type, position]
```

Comment:
You could use deconstruction here: `const [code, type, position] = result` and work with those variables in the log message.

Reply:
I'm familiar with destructuring, I just wasn't sure whether node supported it yet. Is it used somewhere else in the code base?